### PR TITLE
refactor(friends): move friend method from hounddog into getters

### PR DESCRIPTION
**Pull request by jasonwoodland**
**What this PR does** 📖
- Replace $hounddog calls inside components with vuex store getters
- improve action argument typing. will be expanded to other namespaces later

**Which issue(s) this PR fixes** 🔨
AP-1600

**Special notes for reviewers** 🗒️
- maybe these getters can be refactored, but that can happen at a later date. tried to keep this update simple

**Additional comments** 🎤

### CODE REVIEW

**Review by Jekrimo — APPROVED** [review]

File: store/friends/actions.ts [hunk 1]
```diff
@@ -164,7 +169,7 @@ export default {
       ).bytes,
     )
 
-    const friendExists = $Hounddog.friendExists(state, friendKey)
+    const friendExists = getters.friendExists(state)(friendKey)
```
Comment (WanderingHogan) [inline store/friends/actions.ts:172]: this ()() is weird but i like it

**Comment by josephmcg** [comment]
Jason and I collaborated on this one. I plan on building better types for other namespace getters separately

File: store/friends/actions.ts (hunk 1 above)
Comment (josephmcg) [reply store/friends/actions.ts]: I changed this up since the getter doesn't require any state based arguments, it can grab all that on it's own

**Review by phillsatellite — APPROVED** [review]

File: components/views/files/upload/Upload.vue [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <script lang="ts">
 import { FilePlusIcon, PlusIcon } from 'satellite-lucide-icons'
 import Vue, { PropType } from 'vue'
-import { mapState } from 'vuex'
+import { mapState, mapGetters } from 'vuex'
```
Comment (WanderingHogan) [inline components/views/files/upload/Upload.vue:5]: it looks like you took out an un-used computed item but didn't need to replace it with the getters you added?

**Review by WanderingHogan — APPROVED** [review]

File: components/views/files/upload/Upload.vue (hunk 2 above)
Comment (josephmcg) [reply components/views/files/upload/Upload.vue:5]: it was no longer being used in the component. I think it was a leftover from when Samir did a larger refactor of these components

**Review by stavares843 — APPROVED** [review]

**Pull request merged**